### PR TITLE
Re rebase better attribute parser

### PR DESCRIPTION
**Pull request by sebastienbeau**
Rebase #170

### CODE REVIEW

File: lib/locomotive/steam/liquid/tags/concerns/attributes.rb [hunk 1]
```diff
@@ -15,32 +18,113 @@ module Attributes
             private
 
             def parse_attributes(markup, default = {})
-              @attributes     = default || {}
-              @raw_attributes = @attributes.dup
+              @raw_attributes = default.dup || {}
+              attribute_markup = ""
```
Comment (did) [inline lib/locomotive/steam/liquid/tags/concerns/attributes.rb]: single quote

File: lib/locomotive/steam/liquid/tags/concerns/attributes.rb [hunk 2]
```diff
@@ -15,32 +18,113 @@ module Attributes
...
+              if markup =~ /^ *([a-zA-Z0-9_.]*:.*)$/
```
Comment (did) [inline lib/locomotive/steam/liquid/tags/concerns/attributes.rb]: use constants for regexps

File: lib/locomotive/steam/liquid/tags/concerns/attributes.rb [hunk 3]
```diff
@@ -15,32 +18,113 @@ module Attributes
...
+                attribute_markup = $1
+              elsif markup =~ /^[a-zA-Z0-9 _"']*, *(.*)$/
```
Comment (did) [inline lib/locomotive/steam/liquid/tags/concerns/attributes.rb]: use constants for regexps

File: lib/locomotive/steam/liquid/tags/concerns/attributes.rb [hunk 4]
```diff
@@ -15,32 +18,113 @@ module Attributes
...
+              end
+              unless attribute_markup.blank?
+                @raw_attributes.merge!(AttributeParser.parse(attribute_markup))
+              end
+              @raw_attributes
+            end
 
-              return if markup.blank?
+            def context_evaluate_array(vals)
+              vals.map{ value.is_a?(::Liquid::VariableLookup) ? context.evaluate(value) : value }
```
Comment (did) [inline lib/locomotive/steam/liquid/tags/concerns/attributes.rb]: `vals.map {`  (empty space)

File: lib/locomotive/steam/liquid/tags/concerns/path.rb [hunk 5]
```diff
@@ -46,6 +53,18 @@ def render_path(context, &block)
 
             protected
 
+            def parse_compatible_attributes(_attributes)
+              if _attributes
+                _compatible_attribute = _attributes.dup
```
Comment (did) [inline lib/locomotive/steam/liquid/tags/concerns/path.rb]: _compatible_attribute => _compatible_attributes

**Comment by sebastienbeau** [comment]
@did fixed

**Pull request merged**